Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -174,122 +173,109 @@ public class CameraManager {
-                    let interpolatedPitch = interpolator.pitch(at: keyTime)
-
-                    UIView.addKeyframe(withRelativeStartTime: keyTime, relativeDuration: 0.025) {
-                        self.mapView?.cameraView.centerCoordinate = interpolatedCoordinate
-                        self.mapView?.cameraView.zoom = CGFloat(interpolatedZoom)
-                        self.mapView?.cameraView.bearing = CGFloat(interpolatedBearing)
-                        self.mapView?.cameraView.pitch = CGFloat(interpolatedPitch)
-                    }
-                }
-            }
-        }
-
-        if let completion = completion {
-            animator.addCompletion(completion)
+        // Nil out the internalAnimator after `flyTo` finishes
+        flyToAnimator.addCompletion { [weak self](position) in
+            // Call the developer-provided completion (if present)
+            completion?(position)
+            self?.internalAnimator = nil
         }
 
-        animator.startAnimation()
+        flyToAnimator.startAnimation()
+        internalAnimator = flyToAnimator
 
-        return animator
+        return internalAnimator
     }
 
-    /// This function optimizes the bearing for set camera so that it is taking the shortest path.
+    /// Ease the camera to a destination
     /// - Parameters:
-    ///   - startBearing: The current or start bearing of the map viewport.
-    ///   - endBearing: The bearing of where the map viewport should end at.
-    /// - Returns: A `CLLocationDirection` that represents the correct final bearing accounting for positive and negatives.
-    internal func optimizeBearing(startBearing: CLLocationDirection?, endBearing: CLLocationDirection?) -> CLLocationDirection? {
-        // This modulus is required to account for larger values
-        guard
-            let startBearing = startBearing?.truncatingRemainder(dividingBy: 360.0),
-            let endBearing = endBearing?.truncatingRemainder(dividingBy: 360.0)
-        else {
-            return nil
+    ///   - camera: the target camera after animation
+    ///   - duration: duration of the animation
+    ///   - completion: completion to be called after animation
+    /// - Returns: An instance of `CameraAnimatorProtocol` which can be interrupted if necessary
+    @discardableResult
+    public func ease(to camera: CameraOptions, duration: TimeInterval, completion: AnimationCompletion? = nil) -> CameraAnimatorProtocol? {
+
+        internalAnimator?.stopAnimation()
+
+        let animator = makeAnimator(duration: duration, curve: .easeInOut) { (transition) in
+            transition.center.toValue = camera.center
+            transition.padding.toValue = camera.padding
+            transition.anchor.toValue = camera.anchor
+            transition.zoom.toValue = camera.zoom
+            transition.bearing.toValue = camera.bearing
+            transition.pitch.toValue = camera.pitch
```

Comment:
maybe we should add a convenience to set all the toValues with a CameraOptions.

Reply:
I actually quite like the explicitness of the way transitions are defined in the current API. We may want to add this later if we see a need for it.

Reply:
on second thought, i agree with you.. Would be good to have that convenience. Will add it.